diff --color -aur a/src/java.desktop/unix/classes/sun/awt/X11/XWindowPeer.java b/src/java.desktop/unix/classes/sun/awt/X11/XWindowPeer.java
--- a/src/java.desktop/unix/classes/sun/awt/X11/XWindowPeer.java	2024-12-04 02:02:53.000000000 +0100
+++ b/src/java.desktop/unix/classes/sun/awt/X11/XWindowPeer.java	2024-12-05 09:21:52.996521298 +0100
@@ -1260,6 +1260,7 @@
 
     boolean isOverrideRedirect() {
         return XWM.getWMID() == XWM.OPENLOOK_WM ||
+            XWM.getWMID() == XWM.COMPIZ_WM ||
             Window.Type.POPUP.equals(getWindowType());
     }
 
