diff --color -Naur a/src/java.desktop/unix/native/libawt_xawt/awt/gtk_interface.c b/src/java.desktop/unix/native/libawt_xawt/awt/gtk_interface.c
--- a/src/java.desktop/unix/native/libawt_xawt/awt/gtk_interface.c	2024-12-04 02:02:53.000000000 +0100
+++ b/src/java.desktop/unix/native/libawt_xawt/awt/gtk_interface.c	2024-12-06 08:29:44.339921257 +0100
@@ -50,18 +50,18 @@
 
 static GtkLib gtk_libs[] = {
     {
-        GTK_3,
-        JNI_LIB_NAME("gtk-3"),
-        VERSIONED_JNI_LIB_NAME("gtk-3", "0"),
-        &gtk3_load,
-        &gtk3_check
-    },
-    {
         GTK_2,
         JNI_LIB_NAME("gtk-x11-2.0"),
         VERSIONED_JNI_LIB_NAME("gtk-x11-2.0", "0"),
         &gtk2_load,
         &gtk2_check
+    },
+    {
+        GTK_3,
+        JNI_LIB_NAME("gtk-3"),
+        VERSIONED_JNI_LIB_NAME("gtk-3", "0"),
+        &gtk3_load,
+        &gtk3_check
     }
 };
 
